x/stakeibc/keeper/reward_converter.go

```diff
@@ -0,0 +1,442 @@
+	sendTokens := sdk.NewCoin(convertedDenom, amount)
+
+	tradeIcaAddress := route.TradeToHostHop.FromAccount.Address
+	withdrawlIcaAddress := route.TradeToHostHop.ToAccount.Address
+
+	var msgs []proto.Message
+	msgs = append(msgs, &transfertypes.MsgTransfer{
+		SourcePort:       transfertypes.PortID,
+		SourceChannel:    route.TradeToHostHop.TransferChannelId, // channel on tradeZone for transfers to hostZone
+		Token:            sendTokens,
+		Sender:           tradeIcaAddress,
+		Receiver:         withdrawlIcaAddress,
+		TimeoutTimestamp: timeout,
+		Memo:             "",
+	})
+
+	hostZoneId := route.TradeToHostHop.ToAccount.ChainId
+	tradeZoneId := route.TradeToHostHop.FromAccount.ChainId
+	k.Logger(ctx).Info(utils.LogWithHostZone(hostZoneId,
+		"Preparing MsgTransfer of %+v from %s to %s", sendTokens, tradeZoneId, hostZoneId))
+
+	// Send the ICA tx to kick off transfer from hostZone through rewardZone to the tradeZone (no callbacks)
+	tradeZoneConnectionId := route.TradeToHostHop.FromAccount.ConnectionId
+	err := k.SubmitICATxWithoutCallback(ctx, tradeZoneConnectionId, types.ICAAccountType_TRADE, msgs, timeout)
+	if err != nil {
+		return errorsmod.Wrapf(types.ErrICATxFailed, "Failed to submit ICA tx, Messages: %+v, err: %s", msgs, err.Error())
+	}
+
+	return nil
+}
+
+// Trade reward tokens in the Trade ICA for the target output token type using ICA remote tx on trade zone
+// The amount represents the total amount of the reward token in the trade ICA found by the calling ICQ
+// Depending on min and max swap amounts set in the route, it is possible not the full amount given will swap
+func (k Keeper) TradeRewardTokens(ctx sdk.Context, amount sdk.Int, route types.TradeRoute) error {
+	// If the min swap amount was not set it would be ZeroInt, if positive we need to compare to the amount given
+	//  then if the min swap amount is greater than the current amount, do nothing this epoch to avoid small swaps
+	if route.MinSwapAmount.IsPositive() && route.MinSwapAmount.GT(amount) {
+		return nil
+	}
+
+	// If the max swap amount was not set it would be ZeroInt, if positive we need to compare to the amount given
+	//  then if max swap amount is LTE to amount full swap is possible so amount is fine, otherwise set amount to max
+	if route.MaxSwapAmount.IsPositive() && route.MaxSwapAmount.GT(amount) {
+		amount = route.MaxSwapAmount
+	}
+
+	// See if pool swap spot price has been set to a valid ratio (string representing a float like "10.203")
+	// If there is a valid spot price, use it to set a floor for the acceptable minimum output tokens
+	// 5% slippage is allowed so multiply by 0.95 the expected spot price to get the target minimum
+	// minOut is the minimum number of route.TargetDenomOnTradeZone we must receive or the swap will fail
+	minOut := sdk.ZeroInt()
+	if route.SpotPrice != "" {
+		if spotPrice, err := sdk.NewDecFromStr(route.SpotPrice); err == nil {
+			slippageRatio := sdk.NewDecWithPrec(95, 2) // 0.95 to allow 5% loss to slippage
```

Comment:
should we change slippage to be a param on the TradeRoute?

Reply:
Yes but it should be optional and have a default which we specify.  Similar to the min and max swap amounts, if they don't initialize those we can set them to good known defaults.  As discussed on a call, we might want to group all of these params in the route which are rules for how to trade into a new type called TradeConfigs because the update tx route only touches these but would not touch the denoms, hops etc once they are set.

Reply:
Also, we might want to have a variable on here about trade frequency which is essentially the cadence currently hard coded in the stride hourly modulus.  The reason this might need to update, if we lower the max_swap_amount then to move the same volume we would need to speed up the swap cadence.